### ISSUE TITLE
URL validator

### ISSUE THREAD

**Issue by kelunik** [thephpleague/uri#113]
## Issue summary

Given the upcoming changes to https://github.com/thephpleague/uri-parser, which will make it more permissive regarding hosts as defined in the corresponding RFC, there should maybe be a separate package doing validations according to https://url.spec.whatwg.org/, which are specific to URLs, not generic URIs.

I don't have any specific plans, just opening this as a reminder.

**Comment by nyamsprod** [comment]
I understand the idea but in my opinion the parser just split the URI according to general rules that can be applied on any URI/URL string. The validation part should be done based on the URI/URL scheme. That's why URI objects are as important as the parser because they are responsible for the validation part. 
Having a more relax/performant parser as we improve it does not change the fact that URI objects must validate the parser results. So URL requirements should be added/improved directly on the respective URI scheme objects.

This way if https://wiki.php.net/rfc/replace_parse_url where to go into core at some point in the futur we would only need to swap the parser and keep the URI objects.

**Comment by kelunik** [comment]
Yes, totally, that's why I opened the issue here instead of in the `uri-parser` package. Maybe it should have been directly in `uri-schemes`.

**Comment by nyamsprod** [comment]
@kelunik can I close this issue ... as I'm leaning toward deprecating the meta package league/uri in favor of having the league/uri-schemes and the league/uri-components being better promoted ?

**Comment by kelunik** [comment]
@nyamsprod Is the validator for HTTP URIs based on the WHATWG spec?

**Comment by nyamsprod** [comment]
Nope ... Only uses RFC3986 but there's no difference between the two as far as I know or only on edge cases like with Windows path
